frontend/components/forms/admin/AppConfigForm/AppConfigFormFunctional.tsx

```diff
@@ -0,0 +1,1121 @@
+  const handleInputChange = ({ name, value }: IFormField) => {
+    setIterateFormData({ ...iterateFormData, [name]: value });
+  };
+
+  console.log("iterateFormData", iterateFormData);
+
+  // ▀█▀ █▀█ █▀▀ █▀▀ █░░ █▀▀   █▀▄▀█ █▀█ █▀▄ ▄▀█ █░░ █▀
+  // ░█░ █▄█ █▄█ █▄█ █▄▄ ██▄   █░▀░█ █▄█ █▄▀ █▀█ █▄▄ ▄█
+
+  const toggleHostStatusWebhookPreviewModal = () => {
+    setShowHostStatusWebhookPreviewModal(!showHostStatusWebhookPreviewModal);
+    return false;
+  };
+
+  const toggleUsageStatsPreviewModal = () => {
+    setShowUsageStatsPreviewModal(!showUsageStatsPreviewModal);
+    return false;
+  };
+
+  // █▀▀ █▀█ █▀█ █▀▄▀█   █▀ █░█ █▄▄ █▀▄▀█ █ ▀█▀
+  // █▀░ █▄█ █▀▄ █░▀░█   ▄█ █▄█ █▄█ █░▀░█ █ ░█░
+  const onFormSubmit = () => {
+    // Validators
+    const errors: any = {};
+
+    if (enableSSO) {
+      if (!metadata && !metadataURL) {
+        errors.metadata_url = "Metadata URL must be present";
+      }
+      if (!entityID) {
+        errors.entity_id = "Entity ID must be present";
+      }
+      if (!idpName) {
+        errors.idp_name = "Identity Provider Name must be present";
+      }
+    }
+
+    if (!serverURL) {
+      errors.server_url = "Fleet server URL must be present";
+    }
+
+    if (!orgName) {
+      errors.org_name = "Organization name must be present";
+    }
+
+    if (enableSMTP) {
+      if (!smtpSenderAddress) {
+        errors.sender_address = "SMTP sender address must be present";
+      }
+
+      if (!smtpServer) {
+        errors.server = "SMTP server must be present";
+      }
+
+      if (!smtpPort) {
+        errors.server = "SMTP server port must be present";
+      }
+
+      if (smtpAuthenticationType !== "authtype_none") {
+        if (!smtpUsername) {
+          errors.user_name = "SMTP username must be present";
+        }
+
+        if (!smtpPassword) {
+          errors.password = "SMTP password must be present";
+        }
+      }
+    }
+
+    if (enableHostStatusWebhook) {
+      if (!hostStatusWebhookDestinationURL) {
+        errors.destination_url = "Destination URL must be present";
+      }
+
+      if (!hostStatusWebhookHostPercentage) {
+        errors.host_percentage = "Host percentage must be present";
+      }
+
+      if (!hostStatusWebhookDaysCount) {
+        errors.days_count = "Days count must be present";
+      }
+    }
+
+    if (enableHostExpiry) {
+      if (isNaN(hostExpiryWindow) || Number(hostExpiryWindow) <= 0) {
+        errors.host_expiry_window =
+          "Host expiry window must be a positive number";
+      }
+    }
+
+    if (agentOptions) {
+      const { error: yamlError, valid: yamlValid } = validateYaml(agentOptions);
+
+      if (!yamlValid) {
+        errors.agent_options = constructErrorString(yamlError);
+      }
+    }
+
+    setFormErrors(errors);
+
+    if (Object.keys(errors).length !== 0) {
+      return false;
+    }
+
+    // formDataToSubmit mirrors formatting of API not UI
+    const formDataToSubmit = {
```

Comment:
This is painful but unavoidable right now. Previously this was being rebuilt inside of `validate.js`. Can we do that again to cut down on line size?